cas-client-support-saml/src/main/java/org/jasig/cas/client/validation/Saml11TicketValidator.java

```diff
@@ -0,0 +1,238 @@
+            Element attribute;
+            NodeList values;
+            String name;
+            for (int i = 0; i < attributes.getLength(); i++) {
+                attribute = (Element) attributes.item(i);
+                name = attribute.getAttribute("AttributeName");
+                logger.trace("Processing attribute {}", name);
+                values = attribute.getElementsByTagNameNS("*", "AttributeValue");
+                if (values.getLength() == 1) {
+                    principalAttributes.put(name, values.item(0).getTextContent());
+                } else {
+                    final Collection<Object> items = new ArrayList<Object>(values.getLength());
+                    for (int j = 0; j < values.getLength(); j++) {
+                        items.add(values.item(j).getTextContent());
+                    }
+                    principalAttributes.put(name, items);
+                }
+            }
+            return new AssertionImpl(
+                    new AttributePrincipalImpl(nameId, principalAttributes),
+                    assertionValidityStart,
+                    assertionValidityEnd,
+                    new Date(),
+                    Collections.singletonMap(AUTH_METHOD_ATTRIBUTE, (Object) authMethod));
+        } catch (final Exception e) {
+            throw new TicketValidationException("Error processing SAML response", e);
+        }
+    }
+
+    private boolean isValidAssertion(final Date notBefore, final Date notOnOrAfter) {
+        if (notBefore == null || notOnOrAfter == null) {
+            logger.debug("Assertion is not valid because it does not have bounding dates.");
+            return false;
+        }
+
+        final DateTime currentTime = new DateTime(DateTimeZone.UTC);
+        final Interval validityRange = new Interval(
+                new DateTime(notBefore).minus(this.tolerance),
+                new DateTime(notOnOrAfter).plus(this.tolerance));
+
+        if (validityRange.contains(currentTime)) {
+            logger.debug("Current time is within the interval validity.");
+            return true;
+        }
+
+        if (currentTime.isBefore(validityRange.getStart())) {
+            logger.debug("Assertion is not yet valid");
+        } else {
+            logger.debug("Assertion is expired");
+        }
+        return false;
+    }
+
+    protected String retrieveResponseFromServer(final URL validationUrl, final String ticket) {
+        final String request = String.format(
+                SAML_REQUEST_TEMPLATE,
+                generateId(),
+                CommonUtils.formatForUtcTime(new Date()),
+                ticket);
+        HttpURLConnection conn = null;
+        try {
+            conn = this.getURLConnectionFactory().buildHttpURLConnection(validationUrl.openConnection());
+            conn.setRequestMethod("POST");
+            conn.setRequestProperty("Content-Type", "text/xml");
+            conn.setRequestProperty("Content-Type", "text/xml");
```

Comment:
I'll have to review more carefully. I may have intended to set another request header. In any case I'll fix the dupe.